src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+
+The Ingress has to know how to retrieve the Service to then retrieve the Pods and route traffic to them.
+
+The Ingress retrieves the right Service by name and port exposed.
+
+Two things should match in the Ingress and Service:
+
+1. The `servicePort` of the Ingress should match the `port` of the Service
+1. The `serviceName` of the Ingress should match the `name` of the Service
+
+The following diagram summarises the how to connect the ports:
+
+```slideshow
+{
+  "description": "Connecting an Ingress and a Service",
+  "slides": [
+    {
+      "image": "ports-6.svg",
+      "description": "You already know that the Service expose a `port`."
+    },
+    {
+      "image": "ports-7.svg",
+      "description": "The Ingress has a field called `servicePort`."
+    },
+    {
+      "image": "ports-8.svg",
+      "description": "The Service `port` and the Ingress `servicePort` should always match."
+    },
+    {
+      "image": "ports-9.svg",
+      "description": "If you decide to assign port 80 to the service, you should change `servicePort` to 80 too."
+    }
+  ]
+}
+```
+
+In practice, you should look at these lines:
+
+```yaml|highlight=4,7,21,22|title=hello-world.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+  selector:
+    name: app
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: app
+spec:
+  rules:
+  - http:
+    paths:
+    - backend:
+        serviceName: app
+        servicePort: 80
+      path: /
+```
+
+_How do you test that the Ingress works?_
+
+You can use the same strategy as before with `kubectl port-forward`, but instead of connecting to a service, you should connect to the Ingress controller.
+
+First, retrieve the Pod name for the Ingress controller with:
+
+```terminal|command=1|title=bash
+kubectl get pods --all-namespaces
+NAMESPACE   NAME                              READY STATUS
+kube-system coredns-5644d7b6d9-jn7cq          1/1   Running
+kube-system etcd-minikube                     1/1   Running
+kube-system kube-apiserver-minikube           1/1   Running
+kube-system kube-controller-manager-minikube  1/1   Running
+kube-system kube-proxy-zvf2h                  1/1   Running
+kube-system kube-scheduler-minikube           1/1   Running
+kube-system nginx-ingress-controller-6fc5bcc  1/1   Running
+```
+
+Identify the Ingress Pod (which might be in a different Namespace) and describe it to retrieve the port:
+
+```terminal|command=1|title=bash
+kubectl describe pod nginx-ingress-controller-6fc5bcc
+ --namespace kube-system \
+ | grep Ports
+```
+
+Finally, connect to the Pod:
+
+```terminal|command=1|title=bash
+kubectl port-forward nginx-ingress-controller-6fc5bcc 3000:80 --namespace kube-system
+```
+
+At this point, every time you visit port 3000 on your computer, the request is forwarded to port 80 on the Ingress controller Pod.
+
+If you visit <http://localhost:3000>, you should find the app serving a web page.
```

Comment:
I would put a note here that in a real situation wheer you have virtual host names (you ALMOST always have this), then you need to take care of that naming, but maybe you missed this intentionally...

Reply:
I did miss that intentionally. But I have doubts if I should include it now.